lib/src/patterns.dart

```diff
@@ -124,3 +124,73 @@ final tablePattern = RegExp(
 /// A pattern which should never be used. It just satisfies non-nullability of
 /// pattern fields.
 final dummyPattern = RegExp('');
+
+/// A [String] pattern to match a named tag like `<table>` or `</table>`.
+const _namedTagDefinition =
+    // Opening tag begins.
+    '<'
+
+    // Tag name.
+    '[a-z][a-z0-9-]*'
+
+    // Attribute begins, see
+    // https://spec.commonmark.org/0.30/#attribute.
+    r'(?:\s+'
+
+    // Attribute name, see
+    // https://spec.commonmark.org/0.30/#attribute-name.
+    '[a-z_:][a-z0-9._:-]*'
+
+    //
+    '(?:'
+    // Attribute value specification, see
+    // https://spec.commonmark.org/0.30/#attribute-value-specification.
+    r'\s*=\s*'
+
+    // Attribute value, see
+    // https://spec.commonmark.org/0.30/#unquoted-attribute-value.
+    r'''(?:[^\s"'=<>`]+?|'[^']*?'|"[^"]*?")'''
+
+    // Attribute ends.
+    ')?)*'
+
+    // Opening tag ends.
+    r'\s*/?>'
+
+    //
+    '|'
+
+    // Closing tag, see
+    // https://spec.commonmark.org/0.30/#closing-tag.
+    r'</[a-z][a-z0-9-]*\s*>';
+
+final htmlBlockPattern = RegExp(
```

Comment:
Could this be private? Is it referenced anywhere else?

Reply:
Yes, it could be private for now, currently it is only used in `HtmlBlockSyntax`. The idea was to manage all these patterns at one file, also we do not export these patterns.

But it's dependency `_namedTagDefinition` pattern might have another consumer later, I am planing to improve the inline HTML syntax, it's pattern also relay on `_namedTagDefinition`.

Reply:
If there are reasons to share bits between them, sure. If they are only used in one library, better to keep them in one library (and make them private )IMHO